aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -650,6 +658,9 @@ cunn_SoftMaxForward(outscalar_t *output, scalar_t *input, int classes)
 
 template <int ILP, typename scalar_t, typename accscalar_t, typename outscalar_t, template<typename, typename, typename> class Epilogue>
 __global__ void
+#if __CUDA_ARCH__ == 620
+  __launch_bounds__(512, 1)
```

Comment:
The runtime block size for this kernel comes from 
```C++
inline dim3 SoftMax_getBlockSize(int ILP, uint64_t dim_size) {
  uint64_t block_size = 1;
  uint64_t max_block_size = std::min(dim_size / ILP, static_cast<uint64_t>(max_threads));
  ...
```
and [max_threads = 1024](https://github.com/pytorch/pytorch/blob/ebd727fc9f4c1f273b2395b9942cd2dc1d775e93/aten/src/ATen/native/cuda/SoftMax.cu#L102) so to be safe we should use `__launch_bounds__(1024, 1)`.

Reply:
You should also add
```c++
#if __CUDA_ARCH__ == 620
  __launch_bounds__(512, 1)
#endif
```
to [cunn_SoftMaxForward](https://github.com/pytorch/pytorch/blob/ebd727fc9f4c1f273b2395b9942cd2dc1d775e93/aten/src/ATen/native/cuda/SoftMax.cu#L622).